sql/txn.go

```diff
@@ -30,6 +30,9 @@ func (p *planner) BeginTransaction(n *parser.BeginTransaction) (planNode, error)
 // CommitTransaction commits a transaction.
 func (p *planner) CommitTransaction(n *parser.CommitTransaction) (planNode, error) {
 	err := p.txn.Commit()
+	if err != nil {
+		_ = p.txn.Rollback()
```

Comment:
Hmmm, yeah I don't believe it's strictly necessary. I'm taking it out from this PR.